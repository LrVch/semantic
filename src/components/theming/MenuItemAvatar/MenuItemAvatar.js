import {
  Image,
  Menu
} from 'semantic-ui-react'

import React from 'react'
import { device } from '../../../media'
import styled from 'styled-components'

const px = '1.14285714em'

const StyledMenuItemAvatar = styled(Menu.Item)`
  &&& {
    padding-left: ${({ position }) => position === 'right' ? 0 : px};
    padding-right: ${({ position }) => position === 'left' ? 0 : px};

    &:before {
      display: none;
    }

    @media ${device.tablet} {
      padding-left: ${px};
      padding-right: ${px}
    }
  }
`

const MenuItemAvatar = ({
  position = 'right',
  src,
  to = '/'
}) => {
  return (
    <StyledMenuItemAvatar position={position}>
      <Image
        as="a"
        circular
        href={to}
        src={src}
        width={35}
      />
    </StyledMenuItemAvatar>
  )
}

export default MenuItemAvatar
