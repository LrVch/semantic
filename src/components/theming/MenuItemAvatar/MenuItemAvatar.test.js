import 'jest-styled-components'

import { StyledMenuItemAvatar, px } from './MenuItemAvatar'
import { configure, mount, shallow } from 'enzyme'

import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { device } from '../../../media'
import styled from 'styled-components'

configure({ adapter: new Adapter() })

describe('StyledMenuItemAvatar', () => {

  let wrapper

  beforeEach(() => {
    wrapper = mount(<StyledMenuItemAvatar></StyledMenuItemAvatar>)
  })

  test('should create', () => {
    expect(wrapper).toBeTruthy()
  })

  test('should has default styles', () => {
    expect(wrapper).toMatchSnapshot()

    expect(wrapper).toHaveStyleRule('display', 'none', {
      modifier: '&&&:before'
    })

    expect(wrapper).toHaveStyleRule('padding-left', `${px}`, {
      modifier: '&&&',
      media: `${device.tablet}`,
    })

    expect(wrapper).toHaveStyleRule('padding-right', `${px}`, {
      modifier: '&&&',
      media: `${device.tablet}`,
    })

    expect(wrapper).toHaveStyleRule('padding-left', `${px}`, {
      modifier: '&&&'
    })

    expect(wrapper).toHaveStyleRule('padding-right', `${px}`, {
      modifier: '&&&'
    })

    wrapper.setProps({position: 'left'})

    expect(wrapper).toHaveStyleRule('padding-left', `${px}`, {
      modifier: '&&&'
    })

    expect(wrapper).toHaveStyleRule('padding-right', '0', {
      modifier: '&&&'
    })

    wrapper.setProps({position: 'right'})

    expect(wrapper).toHaveStyleRule('padding-left', '0', {
      modifier: '&&&'
    })

    expect(wrapper).toHaveStyleRule('padding-right', `${px}`, {
      modifier: '&&&'
    })
  })
})


