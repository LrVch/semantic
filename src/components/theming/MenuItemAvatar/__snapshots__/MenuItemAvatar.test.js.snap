// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledMenuItemAvatar should has default styles 1`] = `
.c0.c0.c0 {
  padding-left: 1.14285714em;
  padding-right: 1.14285714em;
}

.c0.c0.c0:before {
  display: none;
}

@media (max-width:768px) {
  .c0.c0.c0 {
    padding-left: 1.14285714em;
    padding-right: 1.14285714em;
  }
}

<Styled(MenuItem)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  &&& {
    padding-left: ",
            [Function],
            ";
    padding-right: ",
            [Function],
            ";

    &:before {
      display: none;
    }

    @media ",
            "(max-width: 768px)",
            " {
      padding-left: ",
            "1.14285714em",
            ";
      padding-right: ",
            "1.14285714em",
            "
    }
  }
",
          ],
        },
        "create": [Function],
        "displayName": "Styled(MenuItem)",
        "foldedComponentIds": Array [],
        "handledProps": Array [
          "active",
          "as",
          "children",
          "className",
          "color",
          "content",
          "disabled",
          "fitted",
          "header",
          "icon",
          "index",
          "link",
          "name",
          "onClick",
          "position",
        ],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <MenuItem
      className="c0"
    >
      <div
        className="item c0"
        onClick={[Function]}
      />
    </MenuItem>
  </StyledComponent>
</Styled(MenuItem)>
`;
